pkg/client/gen_test.go

```diff
@@ -393,6 +393,30 @@ func TestGenerateManifestGolden(t *testing.T) {
 				SystemdLogsImage: "custom-systemd-logs:v1.0.0",
 			},
 			goldenFile: filepath.Join("testdata", "custom-systemd-logs-image.golden"),
+		}, {
+			name: "Node selector can be added",
+			inputcm: &client.GenConfig{
+				E2EConfig:     &client.E2EConfig{},
+				Config:        newConfigWithoutUUID(),
+				NodeSelectors: map[string]string{"foo": "bar"},
+			},
+			goldenFile: filepath.Join("testdata", "single-node-selector.golden"),
+		}, {
+			name: "Multiple node selectors can be added",
+			inputcm: &client.GenConfig{
+				E2EConfig:     &client.E2EConfig{},
+				Config:        newConfigWithoutUUID(),
+				NodeSelectors: map[string]string{"foo": "bar", "fizz": "buzz"},
+			},
+			goldenFile: filepath.Join("testdata", "multiple-node-selector.golden"),
+		}, {
+			name: "Node selector with empty string is OK",
```

Comment:
So the `Set` method of the flag should handle that; I added a node_selector_test.go that has a unit test for this. As a result, I removed this test case entirely since it (1) shouldn't occur and (2) produces invalid yaml anyways.